slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c

```diff
@@ -1201,59 +1284,74 @@ int getAllTrimbits() {
 }
 
 enum detectorSettings setSettings(enum detectorSettings sett) {
+    int *dacVals = NULL;
     switch (sett) {
     case STANDARD:
         LOG(logINFOBLUE, ("Setting to standard settings\n"));
-        thisSettings = sett;
-        setDAC(M_VRPREAMP, DEFAULT_STANDARD_VRPREAMP, 0);
-        setDAC(M_VRSHAPER, DEFAULT_STANDARD_VRSHAPER, 0);
+        dacVals = defaultDacValue_standard;
         break;
     case FAST:
         LOG(logINFOBLUE, ("Setting to fast settings\n"));
-        thisSettings = sett;
-        setDAC(M_VRPREAMP, DEFAULT_FAST_VRPREAMP, 0);
-        setDAC(M_VRSHAPER, DEFAULT_FAST_VRSHAPER, 0);
+        dacVals = defaultDacValue_fast;
         break;
     case HIGHGAIN:
         LOG(logINFOBLUE, ("Setting to high gain settings\n"));
-        thisSettings = sett;
-        setDAC(M_VRPREAMP, DEFAULT_HIGHGAIN_VRPREAMP, 0);
-        setDAC(M_VRSHAPER, DEFAULT_HIGHGAIN_VRSHAPER, 0);
+        dacVals = defaultDacValue_highgain;
         break;
     default:
         LOG(logERROR,
             ("Settings %d not defined for this detector\n", (int)sett));
         return thisSettings;
     }
 
+    thisSettings = sett;
+
+    // set special dacs
+    const int specialDacs[] = SPECIALDACINDEX;
+    for (int i = 0; i < NSPECIALDACS; ++i) {
+        setDAC(specialDacs[i], dacVals[i], 0);
+    }
+
     LOG(logINFO, ("Settings: %d\n", thisSettings));
     return thisSettings;
 }
 
 void validateSettings() {
-    if (detectorDacs[M_VRPREAMP] == DEFAULT_STANDARD_VRPREAMP &&
-        detectorDacs[M_VRSHAPER] == DEFAULT_STANDARD_VRSHAPER) {
-        if (thisSettings != STANDARD) {
-            thisSettings = STANDARD;
-            LOG(logINFOBLUE, ("Validated Settings changed to standard!\n"));
-        }
-    } else if (detectorDacs[M_VRPREAMP] == DEFAULT_FAST_VRPREAMP &&
-               detectorDacs[M_VRSHAPER] == DEFAULT_FAST_VRSHAPER) {
-        if (thisSettings != FAST) {
-            thisSettings = FAST;
-            LOG(logINFOBLUE, ("Validated Settings changed to fast!\n"));
+    // if any special dac value is changed individually => undefined
+    const int specialDacs[NSPECIALDACS] = SPECIALDACINDEX;
+    int *specialDacValues[] = {defaultDacValue_standard, defaultDacValue_fast,
+                               defaultDacValue_highgain};
+    int settList[NUMSETTINGS] = {STANDARD, FAST, HIGHGAIN};
```

Comment:
Maybe remove numsettings to be consistent with the line above?

Reply:
done